### PR TITLE
[GEOT-5571] GeometryBuilder.createCurve(PointArray) always make cycled curve

### PR DESCRIPTION
**Pull request by aaryno**
https://osgeo-org.atlassian.net/browse/GEOT-5571

GeometryBuilder.createCurve makes too many points as per GEOT 5571

### CODE REVIEW

**Comment by bencaradocdavies** [comment]
@aaryno I like full Jira titles in pull request and commit titles, and links to and from Jira. A useful trick from @aaime is to Export to XML in Jira to get a nicely formatted title, with Jira tag, ready for copy and paste.

**Comment by tbarsballe** [comment]
Some test failures here @aaryno 
org.geotools.geometry.iso.primitive.SurfaceTest testFastSurfaceFactory and testFastSurfaceBuilder:
java.lang.IllegalArgumentException: Start point of first element has to be at the same position as end point of last element

File: modules/unsupported/geometry/src/test/java/org/geotools/geometry/iso/primitive/CurveTest.java [hunk 1]
```diff
@@ -66,6 +71,7 @@ public void testMain() {
 		
 		this._testCurve(builder);
 		
+
```
Comment (bencaradocdavies) [inline modules/unsupported/geometry/src/test/java/org/geotools/geometry/iso/primitive/CurveTest.java]: Extra blank line?

File: modules/unsupported/geometry/src/test/java/org/geotools/geometry/iso/primitive/CurveTest.java [hunk 2]
```diff
@@ -297,5 +303,18 @@ private void _testCurve(GeometryBuilder builder) {
...
+	@Test 
+	public void testCurveEndsAtLastPoint(){
+            GeometryBuilder builder = new GeometryBuilder(DefaultGeographicCRS.WGS84);
+	    double[] pts = {1.0, 1.0, 0.0, 2.0, 2.0, 0.0, 4.0, 2.0, 1.0, 5.0, 1.0, 1.0 };
+	    PointArray parr = builder.createPointArray(pts);
+	    Curve c=builder.createCurve(parr);
+	    GeometryToString w=new GeometryToString();
+	    System.out.println(w.getString(c));
+            assertTrue(w.getString(c).endsWith("1.0 1.0)"));
+//            assertTrue(w.getString(c).contains("Curve(1.0 1.0, 0.0 2.0, 2.0 0.0, 4.0 2.0, 1.0 5.0, 1.0 1.0)"));
```
Comment (bencaradocdavies) [inline modules/unsupported/geometry/src/test/java/org/geotools/geometry/iso/primitive/CurveTest.java]: Please remove commented code

File: modules/unsupported/geometry/src/test/java/org/geotools/geometry/iso/primitive/CurveTest.java [hunk 3]
```diff
@@ -297,5 +303,18 @@ private void _testCurve(GeometryBuilder builder) {
 
 		
 	}
+	
+	@Test 
+	public void testCurveEndsAtLastPoint(){
+            GeometryBuilder builder = new GeometryBuilder(DefaultGeographicCRS.WGS84);
```
Comment (bencaradocdavies) [inline modules/unsupported/geometry/src/test/java/org/geotools/geometry/iso/primitive/CurveTest.java]: Inconsistent indentation

**Comment by bencaradocdavies** [comment]
Superseded by https://github.com/geotools/geotools/pull/1381 (but might salvage the test).

**Pull request closed without merging**